### ISSUE TITLE
What is the difference between runTask and scheduleTask

### ISSUE THREAD

**Issue by maximusk** [angular/zone.js#810]
Can someone please explain briefly what is the difference between `runTask` and `scheduleTask`? The method descriptions in the sources are not very helpful.

Thanks in advance!

**Comment by JiaLiPassion** [comment]
you can check https://github.com/angular/zone.js/blob/master/doc/task.md this one for more details.

**Comment by maximusk** [comment]
thank you, you decided to follow my suggestion from the previous thread and start with the link? :)

**Comment by JiaLiPassion** [comment]
@maximusk , yeah :-)

**Comment by maximusk** [comment]
good, I'll read it and get back)

**Comment by maximusk** [comment]
@JiaLiPassion , the document you referred to talks about `tasks`. Can I assume that a task is a simply function that is passed to `zone.run`, `zone.scheduleTask` etc?

**Comment by JiaLiPassion** [comment]
@maximusk , the `task` is just the previous question we discussed , `microTask` like `promise`, `macroTask` like `setTimeout`, `eventTask` like `document.addEventListener`, it is the same concept with javascript `task`.

and `Task` is an object which contains the information need to be executed as an async `Task`.
For example.

```
Promise.resolve(5).then(() => {
  // do something.
});
```

The function in `promise.then` will be scheduled as a microTask in javascript `microTaskQueue`, and in `zone.js`, there will be a zone.js version `Task` be scheduled in `zone.js microTaskQueue`.
Basically it is the same lifecycle with `javascript native one`.
